### PR TITLE
refactor:introduce per crate error for batch executors

### LINKED ISSUE #3483: batch: Introduce per crate error for batch executors.

### PR DESCRIPTION
**Pull request by ZENOTME**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
close #3483

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Summarize your change (**mandatory**)
- How does this PR work? Need a brief introduction for the changed logic (optional)
- Describe clearly one logical change and avoid lazy messages (optional)
- Describe any limitations of the current code (optional)

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/batch/src/executor/delete.rs [hunk 1]
```diff
@@ -91,9 +92,7 @@ impl DeleteExecutor {
         let rows_deleted = try_join_all(notifiers)
             .await
             .map_err(|_| {
-                RwError::from(ErrorCode::InternalError(
-                    "failed to wait chunks to be written".to_owned(),
-                ))
+                BatchExecutorError::Internal(anyhow!("failed to wait chunks to be written"))
```
Comment (BowenXiao1999) [inline src/batch/src/executor/delete.rs:95]: Change name to BatchError?

File: src/batch/src/executor/filter.rs [hunk 2]
```diff
@@ -77,14 +78,18 @@ impl FilterExecutor {
                             break;
                         }
                         _ => {
-                            return Err(
-                                InternalError("Data chunk builder error".to_string()).into()
-                            );
+                            return Err(BatchExecutorError::Internal(anyhow!(
+                                "Data chunk builder error"
+                            ))
+                            .into());
```
Comment (BowenXiao1999) [inline src/batch/src/executor/filter.rs:84]: Some times we use ::from, some times use ::Internal? Better unify

**Review by BowenXiao1999 — COMMENTED** [review]
LGTM

**Review by liurenjie1024 — APPROVED** [review]
LGTM

File: src/batch/src/executor/delete.rs (hunk 1 above)
Comment (ZENOTME) [reply src/batch/src/executor/delete.rs:95]: for now, I put error.rs in src/batch/executor/  so I named it BatchExecutorError.
Is good to change the name to BatchError? (there are other crate(batch/execution,batch/rpc) in batch.

Comment (liurenjie1024) [reply src/batch/src/executor/delete.rs:95]: +1 for BatchError, I don't see we need error for each component. And why not move it to src dir?

**Review by BowenXiao1999 — APPROVED** [review]

**Pull request merged**